File: soap/Q2232_ObjectFinalPriceAbsoluteDiscount.java
import soap.q2231.generated.ObjectService;
import soap.q2231.generated.ProductY;
import soap.q2231.generated.SoapObjectService;

public class Q2232_ObjectFinalPriceAbsoluteDiscount {
    static final String STUDENT_CODE = "B21DCCN001";   // TODO
    static final String Q_CODE       = "TODO_qCode";   // TODO

    public static void main(String[] args) {
        ObjectService service = new ObjectService();
        SoapObjectService port = service.getSoapObjectServicePort();

        ProductY p = port.requestProductY(STUDENT_CODE, Q_CODE);
        System.out.printf("price=%.4f tax=%.4f disc=%.4f%n",
                p.getPrice(), p.getTaxRate(), p.getDiscount());

        // discount = SỐ TIỀN tuyệt đối → trừ trực tiếp (KHÁC Q2231/Q2261 dùng %)
        double finalPrice = p.getPrice() * (1 + p.getTaxRate() / 100.0) - p.getDiscount();
        finalPrice = Math.round(finalPrice * 100.0) / 100.0;
        p.setFinalPrice((float) finalPrice);
        System.out.printf("finalPrice=%.2f%n", finalPrice);

        port.submitProductY(STUDENT_CODE, Q_CODE, p);
        System.out.println("Đã submit finalPrice=" + finalPrice);
    }
}

File: soap/Q2232_ObjectFinalPriceAbsoluteDiscount.py
from zeep import Client

EXAM_IP      = "36.50.135.242"
STUDENT_CODE = "B22DCCN863"   # TODO
Q_CODE       = "TODO_qCode"   # TODO
WSDL         = f"http://{EXAM_IP}:2221/ObjectService?wsdl"


def main():
    client = Client(WSDL)

    p = client.service.requestProductY(STUDENT_CODE, Q_CODE)
    print(f"price={p.price:.4f} tax={p.taxRate:.4f} disc={p.discount:.4f}")

    # discount = SỐ TIỀN tuyệt đối → trừ trực tiếp (KHÁC Q2231/Q2261 dùng %)
    final_price = p.price * (1 + p.taxRate / 100.0) - p.discount
    final_price = round(final_price, 2)
    p.finalPrice = float(final_price)
    print(f"finalPrice={final_price:.2f}")

    client.service.submitProductY(STUDENT_CODE, Q_CODE, p)
    print("Đã submit finalPrice=", final_price)


if __name__ == "__main__":
    main()
